src/MongoDB.Driver.Core/Core/Authentication/Sspi/SspiSecurityContext.cs

```diff
@@ -22,38 +22,32 @@ namespace MongoDB.Driver.Core.Authentication.Sspi
     /// <summary>
     /// A wrapper around the SspiHandle structure specifically used as a security context handle.
     /// </summary>
-    internal class SecurityContext : SafeHandle
+    internal class SspiSecurityContext : SafeHandle, ISecurityContext
```

Comment:
I'm not clear on why ISecurityContext doesn't extend SafeHandle.  If GsaapiAuthenticator only knows about ISecurityContext, how will this get disposed?

Reply:
`ISecurityContext` is an interface whereas `SafeHandle` is a base class. I can make `ISecurityContext` extend `IDisposable` and that's probably a good idea. In the current refactoring, I register `SspiSecurityContext` for disposal using a variable of type `SspiSecurityContext` rather than `ISecurityContext`. So I have access to the `Dispose` method on the class definition itself rather than through the interface.